### PR TITLE
[Discover 2.0] Fixed recent query and Data Selector styles

### PR DESCRIPTION
**Pull request by ashwin-pc**
### Description

This PR contains the following changes:
- Fixes recent query to not show duplicates
- Fixes recent query styles
- Fixed Data selector height issue

## Screenshot

https://github.com/user-attachments/assets/6fdbfdee-299f-4972-ba47-0e7a8f84cb82

## Changelog

- skip

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Review by sejli — APPROVED** [review]
LGTM

File: src/plugins/data/public/query/query_string/query_history.ts [hunk 1]
```diff
@@ -4,60 +4,82 @@
  */
 
 import { BehaviorSubject } from 'rxjs';
-import { DataStorage, Dataset } from '../../../common';
+import { DataStorage } from '../../../common';
 import { Query, TimeRange } from '../..';
 
-// Todo: Implement a more advanced QueryHistory class when needed for recent query history
+const MAX_HISTORY_SIZE = 500;
```
[the remark this answers is not in the document]
Comment (ashwin-pc) [reply src/plugins/data/public/query/query_string/query_history.ts:10]: Good call. Fast follow maybe? This was what Abby introduced in her PR but we don't today show any more than 10 to the user anyways, so having this value be user configurable isn't very useful yet. Once we implement more features for the table, then it makes sense adding it as an advanced setting

File: src/plugins/data/public/query/query_string/query_history.ts [hunk 2]
```diff
@@ -4,60 +4,82 @@
...
+        return timeB - timeA; // Sort in descending order (most recent first)
+      });
   }
 
-  getHistory() {
-    return this.getHistoryKeys().map((key) => this.storage.get(key));
+  public getHistory(): any[] {
+    return this.getHistoryKeys()
+      .map((key) => this.storage.get(key))
+      .sort((a, b) => b.time - a.time);
```
Comment (joshuali925) [inline src/plugins/data/public/query/query_string/query_history.ts:34]: getHistoryKeys already sorts by time, is this sort needed?

File: src/plugins/data/public/query/query_string/query_history.ts [hunk 3]
```diff
@@ -4,60 +4,82 @@
...
+import { DataStorage } from '../../../common';
 import { Query, TimeRange } from '../..';
 
-// Todo: Implement a more advanced QueryHistory class when needed for recent query history
+const MAX_HISTORY_SIZE = 500;
+export const HISTORY_KEY_PREFIX = 'query_';
+
 export class QueryHistory {
-  constructor(private readonly storage: DataStorage) {}
+  private changeEmitter: BehaviorSubject<any[]>;
```
Comment (joshuali925) [inline src/plugins/data/public/query/query_string/query_history.ts:14]: i think the type is just
```ts
    interface QueryHistoryItem {
      time: number;
      query: Query;
      dateRange?: TimeRange;
    };
```

**Review by joshuali925 — APPROVED** [review]
nothing blocking

File: src/plugins/data/public/query/query_string/query_history.ts (hunk 2 above)
Comment (ashwin-pc) [reply src/plugins/data/public/query/query_string/query_history.ts:34]: Good catch. I added that sort afterwards. This is no longer needed

File: src/plugins/data/public/query/query_string/language_service/recent_query.tsx [hunk 4]
```diff
@@ -5,165 +5,115 @@
...
-        timeRange: query.timeRange,
-        language: queryLanguage,
-        time: formattedDate,
-      };
+    .filter((item) => item.query.language === currentLanguage)
+    .map((query, idx) => ({
+      id: idx,
+      query: query.query.query,
+      timeRange: query.timeRange,
+      time: moment(query.time).format('MMM D, YYYY HH:mm:ss'),
```
[the remark this answers is not in the document]
Comment (kavilla) [reply src/plugins/data/public/query/query_string/language_service/recent_query.tsx:102]: yeah from ui settings

File: src/plugins/data/public/query/query_string/query_history.ts (hunk 3 above)
Comment (kavilla) [reply src/plugins/data/public/query/query_string/query_history.ts:14]: imo i think the type should extend savedquery

File: src/plugins/data/public/query/query_string/query_history.ts [hunk 5]
```diff
@@ -4,60 +4,82 @@
...
     const subscription = this.changeEmitter.subscribe(listener);
     return () => subscription.unsubscribe();
   }
 
-  addQueryToHistory(query: Query, dateRange?: TimeRange) {
-    const keys = this.getHistoryKeys();
-    keys.splice(0, 500); // only maintain most recent X;
-    keys.forEach((key) => {
-      this.storage.remove(key);
+  public addQueryToHistory(query: Query, dateRange?: TimeRange): void {
```
Comment (kavilla) [inline src/plugins/data/public/query/query_string/query_history.ts:42]: should we just use the LRU library we have?

File: src/plugins/data/public/ui/query_editor/query_editor.tsx [hunk 6]
```diff
@@ -441,7 +450,14 @@ export default class QueryEditorUI extends Component<Props, State> {
           className={classNames('osdQueryEditor__header', this.props.headerClassName)}
         />
         {!this.state.isCollapsed && (
-          <div className="osdQueryEditor__body">{languageEditor.Body()}</div>
+          <>
+            <div className="osdQueryEditor__body">{languageEditor.Body()}</div>
+            <RecentQueriesTable
+              isVisible={this.state.isRecentQueryVisible && useQueryEditor}
+              queryString={this.queryString}
+              onClickRecentQuery={this.onClickRecentQuery}
```
Comment (kavilla) [inline src/plugins/data/public/ui/query_editor/query_editor.tsx:458]: how come not just onClick?

File: src/plugins/data/public/ui/query_editor/query_editor.tsx [hunk 7]
```diff
@@ -441,7 +450,14 @@ export default class QueryEditorUI extends Component<Props, State> {
           className={classNames('osdQueryEditor__header', this.props.headerClassName)}
         />
         {!this.state.isCollapsed && (
-          <div className="osdQueryEditor__body">{languageEditor.Body()}</div>
+          <>
+            <div className="osdQueryEditor__body">{languageEditor.Body()}</div>
+            <RecentQueriesTable
+              isVisible={this.state.isRecentQueryVisible && useQueryEditor}
+              queryString={this.queryString}
```
Comment (kavilla) [inline src/plugins/data/public/ui/query_editor/query_editor.tsx:457]: i think it can access the service with the gettersetters

**Pull request merged**